Proposal2 ML and NLP

ChaiBapchya opened the pull request:
With respect to current requirements and whatever contributions I have made so far, I wish to discuss this proposal. It is an attempt to meet the ongoing project needs. Hope to continue the work that has been just kick-started. @danielballan @dcwalk @b5

ChaiBapchya committed:
Added Proposal for GSoC - Automated Change Monitoring using ML and NLP

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

dcwalk commented:
@ChaiBapchya -- Is this proposal in addition to the one you submitted prior, or meant to replace that one? 

I think we had a chat in the `#gsoc` channel RE: possibility for students to submit multiple proposals (they are!):
> Each student may submit up to 5 proposals to the program. They can submit multiple proposals to the same organization for different projects if they wish.

So just want to make sure this is reviewed in the correct context!

ChaiBapchya commented:
Both the proposals are different. One was about Data storage. This one about processing the data.

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

mhucka commented on GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md:
```diff
@@ -0,0 +1,192 @@
+
+**Project Title:**  
+
+ML+NLP based Automated Change Monitoring
```
Perhaps more accurate would be to say that this is about "ranking" or "prioritization". Monitoring implies even more, while the project scope is actually more of a subset of monitoring.

ChaiBapchya replied:
Yes, agreed.

mhucka commented on GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md:
```diff
@@ -0,0 +1,192 @@
+Major setbacks
+1. Trump Budget announced - Savings of 31% EPA (reduction in fund allocation)
+2. Not a single corpus / dataset on environment can be found online
+3. Absence of an algorithm to track and prioritize changes
```
I think this is irrelevant and actually not quite true :-). You can delete all this, IMHO.

mhucka replied:
(I mean the whole subsection about "major setbacks")

ChaiBapchya replied:
Alright, sure. Thanks

mhucka commented on GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md:
```diff
@@ -0,0 +1,192 @@
+3. Priority Algorithm - 
+
+As currently, there is no standard way of prioritizing which change is to be given precendence over the others, which change is critical and more important than others - an algorithm would be needed for the same. Such an algorithm would be on the basis of "Confidence / Trust factor" and "Rating". Taking these 2 parameters into consideration, every change will be evaluated according to the algorit ...
+It would be a self-learning, ever-evolving process of curation.
```
I can't quite figure out what this is about.

ChaiBapchya replied:
I have tried to explain - 
Sentence
1 - Need of Algo
2,3 - Parameters defining an algo

danielballan replied:
Could you expand on the meaning of those parameters? You don't necessarily have to get this specific but I would suggest either being more specific or less specific.

mhucka commented on GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md:
```diff
@@ -0,0 +1,192 @@
+1. Study PageFreezer API (Webmonitoring processing)
+2. Create corpus of environment data - 
+  + Groundwork - [edgi-govdata-archiving/web-monitoring#28]()
+3. Extracting the content of diff (i.e. Remove html part NLTK - clean_html())
```
Be careful here; it may be necessary to take into consideration some of the information contained in the HTML. For example, if there are changes within an area that is a news headlines section, then the change in the diff probably reflects a change in headlines day-to-day or week-to-week, and not actually a change in meaningful content. So, the bottom line is, it may not be desirable to simply throw away the HTML element tags.

ChaiBapchya replied:
By removing HTML, I mean the element tags. If it is needed, may be I can store the tags somewhere. Basically seperate HTML element tags from the Actual content.

mhucka commented on GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md:
```diff
@@ -0,0 +1,192 @@
+2. Identifying the categories of change
+3. Classification of a change - irrelevant, relevant, important, etc.
+4. Create Primary dataset of changes for ML training
+5. Understanding and Implementing Latent Dirichlet Allocation LDA for identifying Relevant keywords from text
```
I would caution about assuming that a specific machine learning approach should be used, before starting the project. I think it would be better here to have a step of assessing which ML methods are most suitable for the task. LDA could be given as an example, but I'm not sure it makes sense to settle on LDA right now.

ChaiBapchya replied:
Yes, I intended to give LDA as an example itself. Can't finalize one method right now

mhucka commented on GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md:
```diff
@@ -0,0 +1,192 @@
+8. Cross-checking outputs of Summarizer, LDA with Classifier
+9. Fine-tuning the classifier
+10. Designing Classification algorithm to identify the Change type
+11. Studying the existing solutions - 
```
I think it makes sense to study existing solutions, but I'm not sure why it's step #10 here. Wouldn't it make sense to do this earlier? Maybe I misunderstand something.

ChaiBapchya replied:
Yes right. I actually missed the order.

danielballan commented on GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md:
```diff
@@ -0,0 +1,192 @@
+
+**Milestones/Timeline:**  
+
+## Phase 0 - Groundzero
```
You want to use the expression "Square One" here, not "Ground Zero". :- )

ChaiBapchya replied:
Yes, true that.

mhucka commented:
My apologies; this was another case where I never clicked on "submit review" even though I wrote this yesterday.

ChaiBapchya commented:
No issues. Thanks

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

As per the review and suggestions from @mhucka and @danielballan, updated the proposal.

ChaiBapchya commented:
@mhucka @danielballan @dcwalk @b5 @suchthis Have updated the proposal based on the suggestions. Please review. Once approved, I plan to submit it as the final proposal for GSoC (deadline Apr 3) (2 days left)
Awaiting review
Thanks in advance

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

ChaiBapchya committed:
Update GSoC_Proposal_Automated_Change_Monitoring_ML_NLP.md

dcwalk commented:
Thank you for submitting a proposal for Google Summer of Code! 🎉 
I am merging this in as we move into the review phase. Based on [Google's timeline](https://developers.google.com/open-source/gsoc/timeline), on May 4 accepted proposals will be announced

ChaiBapchya commented:
Sure. Thank a lot! It was a pleasure working towards the proposal along with the team!
